File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    
    for num in range(left, right + 1):
        tmp = 0
        for i in range(1, int(num**(1/2)) + 1):
            if num % i == 0:
                if i * i == num:
                    tmp += 1
                else:
                    tmp += 2
        
        if tmp % 2 == 0:
            answer += num
        else:
            answer -= num
    
    return answer

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
function solution(left, right) {
    var answer = 0;
    
    function prime_check(num) {
        let cnt = 0;
        for (let i = 1; i <= parseInt(num ** (1/2)); i++) {
            if (num % i === 0) i * i === num? cnt += 1 : cnt += 2;
        }        
        return cnt % 2 ? false : true;
    }
    
    for (let i = left; i <= right; i++) {
        prime_check(i)? answer += i : answer -= i;
    }
    
    return answer;
}